### PR TITLE
dm-sans: init at 1.002

### PR DESCRIPTION
**Pull request by algorithmiker**
###### Description of changes
It's a geometric sans-serif typeface. More info: https://github.com/googlefonts/dm-fonts

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -4885,6 +4885,11 @@
     githubId = 15957973;
     name = "Jeffry Molanus";
   };
+  gilice = {
+    email = "gilice@proton.me";
+    github = "gilice";
```
Comment (wegank) [inline maintainers/maintainer-list.nix:4890]:
```suggestion
    github = "gilice";
    githubId = 104317939;
```

**Review by wegank — CHANGES_REQUESTED** [review]

File: pkgs/data/fonts/dm-sans/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+  meta = {
+    description =
+      "A geometric sans-serif typeface";
+    homepage =
+      "https://github.com/googlefonts/dm-fonts";
+    license = lib.licenses.ofl;
+    maintainers = with lib.maintainers; [ gilice ];
+  };
+}
```
Comment (wegank) [inline pkgs/data/fonts/dm-sans/default.nix:28]:
```suggestion
}

```

**Review by wegank — CHANGES_REQUESTED** [review]

**Review by wegank — APPROVED** [review]

File: pkgs/data/fonts/dm-sans/default.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
+{ stdenvNoCC, lib, fetchzip }:
+
+stdenvNoCC.mkDerivation rec {
+  pname = "dm-sans";
+  version = "1.002";
+
+  src = fetchzip {
+    name = "${pname}-${version}";
+    url =
+      "https://github.com/googlefonts/dm-fonts/releases/download/v1.002/DeepMindSans_v${version}.zip";
```
Comment (SuperSandro2000) [inline pkgs/data/fonts/dm-sans/default.nix:10]:
```suggestion
    url = "https://github.com/googlefonts/dm-fonts/releases/download/v1.002/DeepMindSans_v${version}.zip";
```

File: pkgs/data/fonts/dm-sans/default.nix [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  installPhase = ''
+    mkdir -p $out/share/fonts/truetype
+    mv *.ttf $out/share/fonts/truetype
+  '';
+
+  meta = {
+    description =
+      "A geometric sans-serif typeface";
+    homepage =
+      "https://github.com/googlefonts/dm-fonts";
```
Comment (SuperSandro2000) [inline pkgs/data/fonts/dm-sans/default.nix:24]:
```suggestion
    description = "A geometric sans-serif typeface";
    homepage = "https://github.com/googlefonts/dm-fonts";
```

File: pkgs/data/fonts/dm-sans/default.nix [hunk 5]
```diff
@@ -0,0 +1,28 @@
+{ stdenvNoCC, lib, fetchzip }:
+
+stdenvNoCC.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/data/fonts/dm-sans/default.nix:3]: We are not compiling anything so we can use fetchzip and postFetch

Comment (gilice) [reply pkgs/data/fonts/dm-sans/default.nix:3]: Thanks, will fix later today.

Comment (zendo) [reply pkgs/data/fonts/dm-sans/default.nix:3]: FYI, You can using `fetchzip` like this:
https://github.com/zendo/nixpkgs/blob/703878418fd940d0770954ba8188f4184340f66f/pkgs/data/fonts/lxgw-wenkai/default.nix

File: pkgs/data/fonts/dm-sans/default.nix [hunk 6]
```diff
@@ -0,0 +1,24 @@
+{ lib, fetchzip }:
+
+fetchzip rec {
+  pname = "dm-sans";
+  version = "1.002";
+  # stripRoot = false;
```
Comment (SuperSandro2000) [inline pkgs/data/fonts/dm-sans/default.nix:6]:
```suggestion
```

File: pkgs/data/fonts/dm-sans/default.nix [hunk 7]
```diff
@@ -0,0 +1,23 @@
+{ lib, fetchzip }:
+
+fetchzip rec {
+  pname = "dm-sans";
+  version = "1.002";
+  url = "https://github.com/googlefonts/dm-fonts/releases/download/v${version}/DeepMindSans_v${version}.zip";
```
Comment (SuperSandro2000) [inline pkgs/data/fonts/dm-sans/default.nix:6]:
```suggestion
  url = "https://github.com/googlefonts/dm-fonts/releases/download/v${version}/DeepMindSans_v${version}.zip";
  stripRoot = false;
```
Apparently this is required but it was commented out so I suggested to remove it.

Comment (gilice) [reply pkgs/data/fonts/dm-sans/default.nix:6]: thanks! due to an error on my side, at the time, I haven't been able to test it locally.
I will correct it. Lets get this finally merged ! :)

File: pkgs/data/fonts/dm-sans/default.nix [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+  stripRoot = false;
+
+  sha256 = "sha256-RSHHxiCac18qqF+hW5M3BbBcra4AQpNLLlUmhiWj9f8=";
+
+  postFetch = ''
+    mkdir -p $out/share/fonts/truetype
+    mv *.ttf $out/share/fonts/truetype
```
Comment (SuperSandro2000) [inline pkgs/data/fonts/dm-sans/default.nix:13]:
```
unpacking source archive /build/DeepMindSans_v1.002.zip
mv: missing destination file operand after '/nix/store/6qj7qpjs6nrs5myb7v1bc6gr78adc52v-dm-sans-1.002/share/fonts/truetype'
Try 'mv --help' for more information.
```

**Comment by gilice** [comment]
hey guys! I finally got around to fixing this. works on my machine now.

**Review by wegank — APPROVED** [review]

**Pull request merged**